### ISSUE TITLE
Remove gSceneryGroundFlags by returning it within GameActionResult

### ISSUE THREAD

**Issue by duncanspumpkin** [OpenRCT2/OpenRCT2#9053]
Almost identical to #8726 but this time for Large and Small Scenery Place Actions. Will require returning a subclass of GameActionResult with the correct additional field.

This will help reduce number of global vars that litter the codebase.

Note this cannot be completed until LargeSceneryPlaceAction is finished #9055.

**Comment by tharkne** [comment]
Hi! I'm not sure if anyone is working on this at the moment, but my team would like to try to take on #9053 if LargeSceneryPlaceAction is to be finished soon. We're new and were hoping this could be a good introduction to the project.

**Comment by duncanspumpkin** [comment]
Sure, give it a shot. It's a tiny bit more complex than 8726 but if you look at that solution you shouldn't be too far off

**Comment by tharkne** [comment]
OK, checking back in with this! Referencing #8726, we started off in LargeSceneryPlaceAction.hpp created a `LargeSceneryPlaceActionResult` subclass with constructors and replaced all the instances of MakeResult() with the appropriate `std::make_unique<LargeSceneryPlaceActionResult>()`.

![image](https://user-images.githubusercontent.com/20932779/56087143-f0a17680-5e32-11e9-8a0d-5eb6127e232a.png)
![image](https://user-images.githubusercontent.com/20932779/56087148-0f077200-5e33-11e9-84b8-01be2eca1bfd.png)

You mentioned in the other thread that for the TrackPlaceAction they needed to edit the callback to `RideConstruction.cpp`, and we expect that we need to do something similar in another file before we can replace the global variable?

**Comment by duncanspumpkin** [comment]
Have a look at all of the users of the class in src/openrct2-ui/windows/TopToolbar.cpp

The ghost one needs an adjustment to use the new class and field.

**Comment by tharkne** [comment]
Ok, I believe all the references to gScecneryGroundFlags regarding Large Scenery placements are now using the subclass GroundFlags field so if the task in SmallSceneryPlaceAction is similar in procedure I think we can get onto that now

**Comment by tharkne** [comment]
Hi, I was testing the scenery placement and noticed that when I try to place Small objects in water that they appear to float on the surface.

![ORST2](https://user-images.githubusercontent.com/20932779/56177757-c20bd300-5fcd-11e9-8e88-edbef881e143.JPG)

I tried replicating this behaviour in the original RCT2 game and it had the scaffolding underneath the scenery for the Small pumpkin object I tested with (the pre-placement preview 'ghost' also did not show the scaffolding), but the cauldron object appeared to float in both games.

![RCT2](https://user-images.githubusercontent.com/20932779/56177605-32febb00-5fcd-11e9-94ef-6aa40b77a643.png)

Is this a known/existing issue in OpenRCT2 or is could have been caused by refactoring the SmallSceneryPlaceAction file?
